### ISSUE TITLE
MongoDB ODM SearchFilter expects DbRef usage on associations, when documentation and lookup code requires raw id

### ISSUE THREAD

**Issue by andytson** [api-platform/core#2574]
The code of the MongoDb ODM SearchFilter expects the association field to be stored as a DbRef, when the documentation states raw ids should be used, and lookups can't use DbRef.
https://github.com/api-platform/core/blob/master/src/Bridge/Doctrine/MongoDbOdm/Filter/SearchFilter.php#L138

Issue present in api-platform/core v2.4.0-beta.2

Steps to reproduce:
* Create a MongoDB BlogPost Document class, and annotate `@ApiFilter(Searchfilter::class, properties={"author": "exact"}` and add annotations for a reference 'author'  to another document Author class with `storeAs="id"`
* Debug the $aggregationBuilder->getPipeline() in SearchFilter
* Make a call to /blogpost/?author=1

Expected pipeline:
```
[{"$match":{"author":{"$in":[1]}}}]
```

Actual pipeline
```
[{"$match":{"author.$id":{"$in":[1]}}}]
```

**Comment by alanpoulain** [comment]
Thank you for your very clear explanation @andytson! I've made a PR to fix it.

**Comment by alanpoulain** [comment]
Should be fixed in the next 2.4 beta.
